app/src/main/java/com/keylesspalace/tusky/components/compose/ComposeViewModel.kt

```diff
@@ -118,29 +120,30 @@ class ComposeViewModel @Inject constructor(
         }
     }
 
-    private fun addMediaToQueue(
+    private suspend fun addMediaToQueue(
         type: QueuedMedia.Type,
         uri: Uri,
         mediaSize: Long,
         description: String? = null
-    ): QueuedMedia {
+    ): QueuedMedia = withContext(Dispatchers.Main) {
+        // running on main dispatcher to avoid race condition when adding multiple media at once, e.g. when restoring from drafts
```

Comment:
how about either `syncrhonized` for id generation or better yet just make an `AtomicInteger` as a field for max id? Seems more robust